### PR TITLE
Add Torus Wagmi Connector

### PR DESCRIPTION
**Pull request by yashovardhan**
This PR adds the source code and demo for Torus Wagmi connector

### CODE REVIEW

File: src/lib/connector.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        await this.torusInstance.init({
+          ...this.torusOptions.TorusParams,
+          network: this.network,
+        });
+      }
+
+      document.getElementById("torusIframe").style.zIndex = "9999999999";
+      await this.torusInstance.login();
+
+      if (this.torusOptions.TorusParams?.showTorusButton) {
```
Comment (himanshuchawla009) [inline src/lib/connector.ts]: fix case of TorusParams

Comment (himanshuchawla009) [reply src/lib/connector.ts]: no need to showTorusButton like this , it will show by default.

File: src/lib/connector.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      const isLoggedIn = await this.isAuthorized();
+
+      // if there is a user logged in, return the user
+      if (isLoggedIn) {
+        const provider = await this.getProvider();
+
+        return {
+          provider,
+          chain: {
+            id: 0,
```
Comment (himanshuchawla009) [inline src/lib/connector.ts]: fix this ID thing, i have fixed in web3auth connector as well.
 Refer: https://github.com/Web3Auth/web3auth-wagmi-connector/pull/4

File: src/lib/connector.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          if (provider.on) {
+            provider.on("accountsChanged", this.onAccountsChanged);
+            provider.on("chainChanged", this.onChainChanged);
+            provider.on("disconnect", this.onDisconnect);
+          }
+
+          return resolve({
+            account,
+            chain: {
+              id: 0,
```
Comment (himanshuchawla009) [inline src/lib/connector.ts]: fix chainId same as above

File: src/lib/connector.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+              unsupported: false,
+            },
+            provider,
+          });
+        }
+        log.error("error while connecting");
+        // eslint-disable-next-line prefer-promise-reject-errors
+        return reject("error");
```
Comment (himanshuchawla009) [inline src/lib/connector.ts]: reject with a better error msg. Also listen to accountChanged event and await on that. isConnected might not be true immediately after login resolves,
so better to listen accountChanged event as well and thn resolve either on isConnected or accountsChanged trigger.

**Comment by himanshuchawla009** [comment]
Test failing:-

1. Login 
2. It will show torus wallet button
3. Logout
4. Login
5. It won't show torus wallet button after login on bottom left, but it should show.

**Comment by yashovardhan** [comment]
@himanshuchawla009 sorry for being late in implementing this fix. It was working for earlier but after clearing cache, it showed the same error as you mentioned.

I just added the fix for it.

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: src/lib/connector.ts (hunk 1 above)
Comment (yashovardhan) [reply src/lib/connector.ts]: We trigger hide button after wagmi connector is disconnected. To remove the error you were facing earlier, this had to be done.

File: src/lib/connector.ts (hunk 2 above)
Comment (yashovardhan) [reply src/lib/connector.ts]: done

File: src/lib/connector.ts (hunk 3 above)
Comment (yashovardhan) [reply src/lib/connector.ts]: done

**Comment by YZhenY** [comment]
can we release this @chaitanyapotti

**Review by chaitanyapotti — APPROVED** [review]
I see that my comments were implemented. lgtm

**Pull request merged**